### DISCUSSION TITLE
Any ideas for a global message box?

### DISCUSSION THREAD

**Discussion by CarbonPool** [slint-ui/slint#5046] [Q&A]
How to create a message box in any situation using rust code and dynamically execute different task callbacks.

**Comment by hunger** [comment]
What kind of entities do you want to have communicating? Is this in-porcess or inter-process?

For in-process (some parts of your program sending messages to other parts of your program) you might be able to get away with a simple [mpsc channel](https://doc.rust-lang.org/std/sync/mpsc/fn.channel.html) -- at least in the simple case. There are other crates out there that support more complex use-cases if that is necessary.

If you are thinking about processes asking other processes to do something on their behalf, you might want to look into something that like the [zbus](https://docs.rs/zbus/latest/zbus/) crate -- at least on Linux. On Android you might look into intends, I can not think of the names of similar mechanismns on Mac and Windows right now though:-)
